src/libraries/System.Private.CoreLib/src/System/Threading/SemaphoreSlim.cs

```diff
@@ -341,16 +341,10 @@ public bool Wait(int millisecondsTimeout, CancellationToken cancellationToken)
                         }
                     }
                 }
-                // entering the lock and incrementing waiters must not suffer a thread-abort, else we cannot
-                // clean up m_waitCount correctly, which may lead to deadlock due to non-woken waiters.
-                try { }
-                finally
+                Monitor.Enter(m_lockObjAndDisposed, ref lockTaken);
+                if (lockTaken)
```

Comment:
This does not need `if (lockTaken)` because `lockTaken` is always true if [Monitor.Enter](https://docs.microsoft.com/dotnet/api/system.threading.monitor.enter?view=net-5.0#System_Threading_Monitor_Enter_System_Object_System_Boolean__) returns rather than throws.

Reply:
Thanks. Might be a reasonable Roslyn analyzer.